sdcm/sct_config.py

```diff
@@ -1678,13 +1678,36 @@ def __init__(self):
             backend_config_files += self.defaults_config_files[str(backend)]
         self.multi_region_params = self.per_provider_multi_region_params.get(str(backend), [])
 
+        def _load_merge_strategy(
+                self, other: Dict[str, Any], key: str, val: Any = None, **options) -> None:
+            """
+            Custom merge strategy for anyconfig to merge 'append_scylla_yaml' attribute values from multiple configs.
+
+            :param self: mapping object to update with `other`
+            :param other: mapping object to update `self`
+            :param key: key of mapping object to update
+            :param val: value to update self alternatively
+            """
+            if key == 'append_scylla_yaml':
+                if key in self and key in other:
+                    self[key] += other[key]
```

Comment:
I'm sure pure string append would be enough in this case....

maybe we need to parse it from yaml, and then merge and turn it back to string ?

what happens if we'll have multiple values ?

i.e. keep in mind `append_scylla_yaml` is currently defined as string, that holds a yaml, and not part of the yaml

Reply:
regardless I think this specific change might need a unit test or a few of them
proving how it would work

Reply:
>I'm sure pure string append would be enough in this case....

Not sure, maybe I'm missing something from the comment.
The `self[key] += other[key]` exactly appends value of `append_scylla_yaml` from one config (config that is being merged) to the value of the other (config to which merge is performed)

Reply:
>regardless I think this specific change might need a unit test or a few of them
proving how it would work

I'll unit test. Test run on real test configurations show that merging the value works as expected.

Reply:
build a test that both config file has the same key in `append_scylla_yaml`, and you'll get a broken yaml inside of it

config1
```
append_scylla_yaml: |
  enable_sstables_mc_format: true
  enable_sstables_md_format: true
```

config2
```
append_scylla_yaml: |
  enable_sstables_mc_format: false
  enable_sstables_md_format: false
```

the outcome would be:

```
append_scylla_yaml: |
  enable_sstables_mc_format: true
  enable_sstables_md_format: true
  enable_sstables_mc_format: false
  enable_sstables_md_format: false
```

wouldn't it ?

Reply:
we need the ability to overwrite things inside the yaml **IN** `append_scylla_yaml` value

Reply:
we can change the definition of `append_scylla_yaml` to stop being a string, and make it a first class yaml structure.
but we'll need to carefully attend to the 68 usages of it while doing so. (it's not a very big change, but need to be careful)

Reply:
@soyacz saying the same https://github.com/scylladb/scylla-cluster-tests/pull/7554#issuecomment-2146955643

with with the exact details and example of other config option handling the same situation.

Reply:
Ok, so we don't want to be able to merge values of `append_scylla_yaml` attribute as they are used now in the STC.
But to rework how this attribute is used and get rid of defining its value as a multiline string everywhere in SCT?

Reply:
got it, will be back soon